### PR TITLE
Ports plasmaman departmental envirosuits and tweaks.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Ports /tg/station PRs #42245,  #42859, #43088,  #43101, a tiny bit of #43317, #44881, and #45369.
Also personally cut down the number of envirohelmet's "lights on" icon_states in favour of an applied overlay, and head of staff envirosuits since every flake can (sadly) be captain here.

Ready.

## Why It's Good For The Game
Porting in some plasmameme updates from /tg/station.

## Changelog
:cl: Ghommie (original PRs by subject217, AarontheIdiot, pireamaineach, Gousaid67 and SouDescolado)
balance: Removed plasmamen species speedmod in favor of a slowdown for envirosuits.
add: Nanotrasen has began deploying departementalized enviro plasmasuits to the station! plasmamens can now benefit from some of the bonuses aswell as the color pattern of their job, while allowing others to easily determine their profession!
fix: Benevolent Nanotrasen makes gulag available for everyone! (Plasmamen retain their equipment and don't die.)
del: Removes code that theoretically limits plasmamen from being clowns and mimes, but actually doesn't.
/:cl:

### CODE REVIEW

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 1]
```diff
@@ -41,16 +41,23 @@
 	icon_state = "plasmaman-helm"
 	item_state = "plasmaman-helm"
 	strip_delay = 80
+	flash_protect = 0
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/spacesuits/plasmamen.dm]: !
Leave the flash protection 1, unlike tg and this pr they still get heavy nerfs under the hood for quite a few things.

Comment (Ghommie) [reply code/modules/clothing/spacesuits/plasmamen.dm]: This change is legit and not unlisted, but if the masses argue against it, I can undo this bit.

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 2]
```diff
@@ -61,3 +68,166 @@
...
+	desc = "A finely tailored azure envirohelm designed for head of personell."
+	icon_state = "hop_envirohelm"
+	item_state = "hop_envirohelm"
+
+/obj/item/clothing/head/helmet/space/plasmaman/captain
+	name = "captain's plasma envirosuit helmet"
+	desc = "A blue and gold envirohelm designed for the station's captain, nonetheless. Made of superior materials to protect them from the station hazards and more."
+	icon_state = "captain_envirohelm"
+	item_state = "captain_envirohelm"
+	armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 10, "bomb" = 10, "bio" = 100, "rad" = 10, "fire" = 100, "acid" = 85)
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/spacesuits/plasmamen.dm]: Dont give them Massively good armor that cap dosnt normaly get

File: code/modules/mob/living/carbon/human/species_types/plasmamen.dm [hunk 3]
```diff
@@ -16,7 +16,6 @@
 	burnmod = 1.5
 	heatmod = 1.5
 	breathid = "tox"
-	speedmod = 1
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/human/species_types/plasmamen.dm:19]: Keep the speed mod, its their Race Slow Down they get

**Comment by lolman360** [comment]
>flashproofing changes 

NO NO NO NO NO

**Review by silicons — APPROVED** [review]

**Pull request merged**